felix/statusrep/status_file_reporter.go

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statusrep
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/projectcalico/calico/felix/deltatracker"
+	"github.com/projectcalico/calico/felix/proto"
+	"github.com/projectcalico/calico/libcalico-go/lib/names"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	dirPolicyStatus = "policy"
+)
+
+// EndpointStatusFileReporter writes a file to the FS
+// any time it sees an Endpoint go up in the dataplane.
+//
+//   - Currently only writes to a directory "policy", creating
```

Comment:
Sure, can just make it "status". My outlook was to name the directory as if it might have other forms of statuses, besides policy status in the future. If we're happy conflating policy-status with "status" then I'll consolidate that change across all the config params

Reply:
Yep, that's my take, this is the done with endpoint high-level status.  Can add more statuses later if needed.  I think I'd probably add content to the file at that point.  Make it a JSON blob with more detail in it.

Reply:
Still referencing policy here?